content/docs/tasks/traffic-management/secure-ingress.md

```diff
@@ -0,0 +1,285 @@
+with a certificate and a private key. Then we create a `Gateway` definition that contains a `server` on the port 443.
+
+1. Create a Kubernetes `Secret` to hold the server's certificate and private key. Use `kubectl` to create the secret
+`istio-ingressgateway-certs` in namespace `istio-system` . The Istio gateway will load the secret automatically.
+
+    > The secret MUST be called `istio-ingressgateway-certs` in the `istio-system` namespace, or it will not
+    > be mounted and available to the Istio gateway.
+
+    ```command
+    $ kubectl create -n istio-system secret tls istio-ingressgateway-certs --key 3_application/private/httpbin.example.com.key.pem --cert 3_application/certs/httpbin.example.com.cert.pem
+    secret "istio-ingressgateway-certs" created
+    ```
+
+    Note that by default all the service accounts in the `istio-system` namespace can access this secret, so the private
+    key can be leaked. You can change the
+    [Role-Based Access Control (RBAC)](https://kubernetes.io/docs/reference/access-authn-authz/rbac/) rules to protect
+    it.
+
+1.  Define a `Gateway` with a `server` section for the port 443.
+
+    > The location of the certificate and the private key MUST be `/etc/istio/ingressgateway-certs`, or the gateway will fail to load them.
+
+    ```bash
+        cat <<EOF | istioctl create -f -
+        apiVersion: networking.istio.io/v1alpha3
+        kind: Gateway
+        metadata:
+          name: httpbin-gateway
+        spec:
+          selector:
+            istio: ingressgateway # use istio default ingress gateway
+          servers:
+          - port:
+              number: 443
+              name: https
+              protocol: HTTPS
+            tls:
+              mode: SIMPLE
+              serverCertificate: /etc/istio/ingressgateway-certs/tls.crt
+              privateKey: /etc/istio/ingressgateway-certs/tls.key
+            hosts:
+            - "httpbin.example.com"
+        EOF
+    ```
+
+1.  Configure routes for traffic entering via the `Gateway`. Define the same `VirtualService` as in the [Control Ingress Traffic](/docs/tasks/traffic-management/ingress/#configuring-ingress-using-an-istio-gateway) task:
+
+    ```bash
+        cat <<EOF | istioctl create -f -
+        apiVersion: networking.istio.io/v1alpha3
+        kind: VirtualService
+        metadata:
+          name: httpbin
+        spec:
+          hosts:
+          - "httpbin.example.com"
+          gateways:
+          - httpbin-gateway
+          http:
+          - match:
+            - uri:
+                prefix: /status
+            - uri:
+                prefix: /delay
+            route:
+            - destination:
+                port:
+                  number: 8000
+                host: httpbin
+        EOF
+    ```
+
+1.  Access the _httpbin_ service by HTTPS, sending an HTTPS request by _curl_ to `SECURE_INGRESS_PORT`.
```

Comment:
You deleted from the ingress task the explanation of what --resolve is for:

**Note the `--resolve` flag that we use this time. Unlike HTTP requests, the `Host` header that we pass will be encrypted this time, so the ingress gateway will not be able to use it to match the request to our configuration. The `--resolve` flag instructs _curl_ to supply the [SNI](https://en.wikipedia.org/wiki/Server_Name_Indication) value "httpbin.example.com" when accessing the gateway IP over TLS.**

It would be good to work something similar into this task.

Reply:
@frankbu Good catch, it was not my intention. I will put this part back, rewriting it to use "you" instead "we".

Reply:
@frankbu I put a comment about `--resolve` flag, please review.

Reply:
I would reword this step a little, and break up the text like this:

1. Access the _httpbin_ service with HTTPS by sending an `https` request using _curl_ to `SECURE_INGRESS_PORT`. 

   The --resolve flag is used to instruct _curl_ to supply the [SNI](https://en.wikipedia.org/wiki/Server_Name_Indication) value "httpbin.example.com" when accessing the gateway IP over TLS. The `--cacert` option instructs _curl_ to use your generated certificate to verify the server. 

   By sending the request to the `/status/418` URL path, you get a nice visual clue that your _httpbin_ service was indeed accessed. The _httpbin_ service will return the [418 I'm a Teapot](https://tools.ietf.org/html/rfc7168#section-2.3.3) code.

Reply:
@frankbu Done, thanks.